packages/database/src/migrations/20210118040638-AddFetpProject-modifies-data.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+import { codeToId, insertObject, generateId } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const countryCodes = ['PG', 'SB'];
+const projectCode = 'fetp';
+const projectName = 'FETP';
+const projectDescription = 'Field Epidemiologist Training Program Graduate Information';
+
+const dashboardGroupName = 'FETP';
+const projectDashboardReports = '{project_details}';
+const userGroups = ['Public', 'FETP Graduates'];
+
+const getProjectDashboardGroupCode = userGroup => `FETP_Project_${userGroup.split(' ').join('_')}`;
+
+const getCountryDashboardGroupCode = (countryCode, userGroup) =>
+  `${countryCode}_FETP_Country_${userGroup.split(' ').join('_')}`;
+
+export const hierarchyNameToId = async (db, name) => {
+  const record = await db.runSql(`SELECT id FROM entity_hierarchy WHERE name = '${name}'`);
+  return record.rows[0] && record.rows[0].id;
+};
+
+const addCountryToProject = async (db, countryCode) => {
+  await insertObject(db, 'entity_relation', {
+    id: generateId(),
+    parent_id: await codeToId(db, 'entity', projectCode),
+    child_id: await codeToId(db, 'entity', countryCode),
+    entity_hierarchy_id: await hierarchyNameToId(db, projectCode),
+  });
+  await Promise.all(
+    userGroups.map(userGroup =>
+      insertObject(db, 'dashboardGroup', {
+        organisationLevel: 'Country',
+        userGroup,
+        organisationUnitCode: countryCode,
+        dashboardReports: '{project_details}',
+        name: dashboardGroupName,
+        code: getCountryDashboardGroupCode(countryCode, userGroup),
+        projectCodes: `{${projectCode}}`,
+      }),
+    ),
+  );
+};
+
+exports.up = async function (db) {
+  await Promise.all(
+    userGroups.map(userGroup =>
+      insertObject(db, 'dashboardGroup', {
+        organisationLevel: 'Project',
+        userGroup,
+        organisationUnitCode: projectCode,
+        dashboardReports: projectDashboardReports,
+        name: projectName,
+        code: getProjectDashboardGroupCode(userGroup),
+        projectCodes: `{${projectCode}}`,
+      }),
+    ),
```

Comment:
Is this a special case to add multiple `userGroup` in `orgnisationLevel: Project`? Correct me if I'm wrong, I guess the `userGroup` here usually is the most public one?

Reply:
Thanks @billli0, originally I was thinking from comment ' Basic project information can be available to the Public', that Public on project would be description dashboard and regional under 'FETP Graduates', but having a look through the cards looks like all reports are at country level, so I think you are correct and I'll limit project dashboard to public.